### PR TITLE
Implemented state remove command

### PR DESCRIPTION
**Pull request by aleksanderaleksic**
Implemented the state remove command, with a lot of inspiration from the state move command

### CODE REVIEW

**Comment by paultyng** [comment]
@aleksanderaleksic this mostly looks good to me, do you think you could also add an e2e test (see the `tfexec/internal/e2etest` directory). That will at least let it get exercised across multiple versions of TF to ensure its present and works in an expected way.

**Comment by aleksanderaleksic** [comment]
@paultyng Added a test now 👍🏻

**Review by paultyng — APPROVED** [review]
LGTM, would like to have @kmoe give a once over as well though.

File: tfexec/state_rm.go [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+func (opt *StateOption) configureStateRm(conf *stateRmConfig) {
+	conf.state = opt.path
+}
+
+func (opt *StateOutOption) configureStateRm(conf *stateRmConfig) {
+	conf.stateOut = opt.path
+}
+
+// StateRm represents the terraform state rm subcommand.
+func (tf *Terraform) StateRm(ctx context.Context, source string, opts ...StateRmCmdOption) error {
```
Comment (kmoe) [inline tfexec/state_rm.go:58]: `address` would be a clearer name for this parameter than `source` here, I think, corresponding to the cmd help text:

```
$ terraform -help state rm
Usage: terraform state rm [options] ADDRESS...
```

**Review by kmoe — APPROVED** [review]
Thank you, looks good. One comment regarding variable naming.

File: tfexec/state_rm.go (hunk 1 above)
Comment (aleksanderaleksic) [reply tfexec/state_rm.go:58]: Yep, I agree will fix 👍🏻

**Pull request merged**